GVFS/GVFS.Common/NuGetUpgrade/ModernNuGetUpgrader.cs

```diff
@@ -0,0 +1,256 @@
+
+            upgrader = new ModernNuGetUpgrader(
+                ProcessHelper.GetCurrentProcessVersion(),
+                tracer,
+                fileSystem,
+                httpClient,
+                dryRun,
+                noVerify,
+                upgraderConfig,
+                ProductUpgraderInfo.GetAssetDownloadsPath(),
+                platform,
+                credentialStore);
+
+            return true;
+        }
+
+        public override bool TryQueryNewestVersion(out Version newVersion, out string message)
+        {
+            newVersion = null;
+            string orgName = this.OrgName;
+
+            if (orgName == null)
+            {
+                message = "ModernNuGetUpgrader is not able to parse org name from NuGet Package Feed URL";
+                return false;
+            }
+
+            OrgInfoApiClient infoServer = new OrgInfoApiClient(this.httpClient, this.OrgInfoServerUrl);
+
+            try
+            {
+                this.highestVersionAvailable = infoServer.QueryNewestVersion(orgName, this.platform, this.Ring);
+            }
+            catch (HttpRequestException exception)
+            {
+                message = string.Format("Network error: could not connect to server ({0}). {1}", this.OrgInfoServerUrl, exception.Message);
+                this.TraceException(exception, nameof(this.TryQueryNewestVersion), $"Error connecting to server.");
+
+                return false;
+            }
+            catch (TaskCanceledException exception)
+            {
+                // GetStringAsync can also throw a TaskCanceledException to indicate a timeout
+                // https://github.com/dotnet/corefx/issues/20296
+                message = string.Format("Network error: could not connect to server ({0}). {1}", this.OrgInfoServerUrl, exception.Message);
+                this.TraceException(exception, nameof(this.TryQueryNewestVersion), $"Error connecting to server.");
+
+                return false;
+            }
+            catch (SerializationException exception)
+            {
+                message = string.Format("Parse error: could not parse response from server({0}). {1}", this.OrgInfoServerUrl, exception.Message);
+                this.TraceException(exception, nameof(this.TryQueryNewestVersion), $"Error parsing response from server.");
+
+                return false;
+            }
+
+            if (this.highestVersionAvailable != null &&
+                this.highestVersionAvailable > this.installedVersion)
+            {
+                newVersion = this.highestVersionAvailable;
+            }
+
+            if (newVersion != null)
+            {
+                this.tracer.RelatedInfo($"{nameof(this.TryQueryNewestVersion)} - new version available: installedVersion: {this.installedVersion}, highestVersionAvailable: {newVersion}");
+                message = $"New version {this.highestVersionAvailable} is available.";

```

Comment:
Shouldn't this be using `newVersion` since that is what is getting the null check?  How do we know `this.highestVersionAvailable` isn't null here?

Reply:
yeah - `newVersion` is probably more clear.

> How do we know this.highestVersionAvailable isn't null here?

 if `newVersion` is not null, then `newVersion == highestVersionAvailable`..

I will update to `newVersion`